### PR TITLE
Remove AnonymizerResult in favor of OperatorResult

### PR DESCRIPTION
**Pull request by SharonHart**
## Change Description

Briefing through the code,  I've noticed that the only plqce where AnonymizerResult is referenced, is as the input for the deanonymize method. It is no longer the anonymizer result. Apart from naming being off, I saw that the input can be the actual result from the anonymize method, which simplifies 
the end-to-end flow between anonymization and deanonymzation.  

```python
anonymize_result = engine.anonymize(
    text=text,
    analyzer_results=[
        RecognizerResult(entity_type="PERSON", start=11, end=21, score=0.8),
    ],
    operators={"PERSON": OperatorConfig("encrypt", {"key": crypto_key})},
)

#Before the change
DeanonymizeEngine().deanonymize(
    text=anonymize_result.text,
    entities=AnonymizerResult(start=11, end=55, entity_type="PERSON"), # Construct from anonymize_result.items in a for loop
    operators={"DEFAULT": OperatorConfig("decrypt", {"key": crypto_key})},
)


#After the change
DeanonymizeEngine().deanonymize(
    text=anonymize_result.text,
    entities=anonymize_result.items,
    operators={"DEFAULT": OperatorConfig("decrypt", {"key": crypto_key})},
)
```
This change is more native to presidio component integration and similar to analyzer-anonymizer input-output transmission. 

I'm not sure, but this might was the intention at first place? Maybe stopped at the middle?

What's left (after discussing if the solution is applicable):
- [x] Fix tests
- [x] Fix docs, samples accordingly  
- [ ] Changelog

### CODE REVIEW

**Review by omri374 — COMMENTED** [review]
Looks great. Thanks.
This is a (small) API change, isn't it? If yes, let's bump the next minor version + could you please add the change to the change log?

**Review by omri374 — APPROVED** [review]

File: presidio-anonymizer/presidio_anonymizer/deanonymize_engine.py [hunk 1]
```diff
@@ -3,8 +3,7 @@
 from typing import List, Dict
 
 from presidio_anonymizer.core.engine_base import EngineBase
-from presidio_anonymizer.entities import OperatorConfig
-from presidio_anonymizer.entities.engine.anonymizer_result import AnonymizerResult
+from presidio_anonymizer.entities import OperatorConfig, OperatorResult
 from presidio_anonymizer.entities.engine.result.engine_result import EngineResult
```
[the remark this answers is not in the document]
Comment (SharonHart) [reply presidio-anonymizer/presidio_anonymizer/deanonymize_engine.py:7]: For EngineResult? Not my change but I'll try

**Comment by SharonHart** [comment]
@omri374 The only change is the anonymizer input object. I wouldn't say that this is a small API change since for those who already use deanonymizer in the package usage - it's a breaking change (it would, however, simplify).

File: presidio-anonymizer/presidio_anonymizer/core/engine_base.py [hunk 2]
```diff
@@ -60,11 +60,11 @@ def _operate(
             # ordered from end to start, and the indexes will be normalized
             # from start to end once the loop ends and the text length is deterministic.
```
Comment (shiranr) [inline presidio-anonymizer/presidio_anonymizer/core/engine_base.py:61]: The version of this change should be a breaking one :)

**Review by shiranr — DISMISSED** [review]
Looks good, 2 small comments

**Review by navalev — APPROVED** [review]

**Review by omri374 — APPROVED** [review]

**Pull request merged**